store/tikv/kv.go

```diff
@@ -198,6 +208,44 @@ func (s *KVStore) BeginWithExactStaleness(txnScope string, prevSec uint64) (*KVT
 	return txn, nil
 }
 
+// BeginWithMinStartTS begins transaction with the least startTS
+func (s *KVStore) BeginWithMinStartTS(txnScope string, minStartTS uint64) (*KVTxn, error) {
+	stores := make([]*Store, 0)
+	allStores := s.regionCache.getStoresByType(tikvrpc.TiKV)
+	if txnScope != oracle.GlobalTxnScope {
+		for _, store := range allStores {
+			if store.IsLabelsMatch([]*metapb.StoreLabel{
+				{
+					Key:   "zone",
+					Value: txnScope,
+				},
+			}) {
+				stores = append(stores, store)
+			}
+		}
+	} else {
+		stores = allStores
+	}
+	resolveTS := s.getMinResolveTSByStores(stores)
+	startTS := minStartTS
+	// If the resolveTS is larger then then minStartTS, we will use resolveTS as StartTS, otherwise we will use
+	// minStartTS directly.
+	if oracle.CompareTS(startTS, resolveTS) < 0 {
+		startTS = resolveTS
+	}
+	return s.BeginWithStartTS(txnScope, startTS)
+}
+
+// BeginWithMaxPrevSec begins transaction with given max previous seconds for startTS
+func (s *KVStore) BeginWithMaxPrevSec(txnScope string, maxPrevSec uint64) (*KVTxn, error) {
+	bo := NewBackofferWithVars(context.Background(), tsoMaxBackoff, nil)
+	minStartTS, err := s.getStalenessTimestamp(bo, txnScope, maxPrevSec)
```

Comment:
Why not compare with resolvedTS?

Reply:
`BeginWithMaxPrevSec` will directly call `BeginWithMinStartTS` after getting minStartTS, and the reslovedTS will be compared with minStartTS in `BeginWithMinStartTS`